picard/coverart/providers/__init__.py

```diff
@@ -17,23 +17,74 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
+from picard import log, config
 from picard.plugin import ExtensionPoint
+from picard.ui.options import OptionsPage, register_options_page
 
 
 _cover_art_providers = ExtensionPoint()
 
 
+class ProviderOptions(OptionsPage):
+
+    """
+        Template class for provider's options
+    """
+
+    PARENT = "cover"
+    SORT_ORDER = 0
+    ACTIVE = True
+
+    options = []
+
+    _options_ui = None
+
+    def __init__(self, parent=None):
+        super(ProviderOptions, self).__init__(parent)
+        if callable(self._options_ui):
```

Comment:
If `_options_ui` is not callable the user/developer of a subclass of this has misused this class, that should definitely be an exception.

This also doesn't seem to be very specific to cover art provider options pages - all option pages need to initialize their UI somehow. Do we want to move this up to `OptionsPage`?